pydis_site/apps/main/tests/test_repodata_helpers.py

```diff
@@ -0,0 +1,66 @@
+from datetime import timedelta
+
+from django.conf import settings
+from django.test import TestCase
+from django.utils import timezone
+
+from pydis_site.apps.main.models import RepositoryMetadata
+from pydis_site.apps.main.views import HomeView
+
+
+class TestRepositoryMetadataHelpers(TestCase):
+
+    def test_returns_metadata(self):
+        """Test if the _get_repo_data helper actually returns what it should."""
+
+        home_view = HomeView()
+        metadata = home_view._get_repo_data()
+
+        self.assertIsInstance(metadata[0], RepositoryMetadata)
+        self.assertEquals(len(metadata), len(settings.HOMEPAGE_REPOS))
+
+    def test_returns_cached_metadata(self):
+        """Test if the _get_repo_data helper returns cached data when available."""
+
+        home_view = HomeView()
+        repo_data = RepositoryMetadata(
+            repo_name="python-discord/site",
+            description="testrepo",
+            forks=42,
+            stargazers=42,
+            language="English",
+        )
+        repo_data.save()
+        metadata = home_view._get_repo_data()
+
+        self.assertIsInstance(metadata[0], RepositoryMetadata)
+        print(metadata[0])  # Tests the __str__ in the model
```

Comment:
how about doing something similar to the tests for the api models, just checking that
```suggest
self.assertIsInstance(str(metadata[0]), str)
```